pkg/controller/operators/catalog/subscription/reconciler.go

```diff
@@ -109,6 +122,105 @@ func (c *catalogHealthReconciler) Reconcile(ctx context.Context, in kubestate.St
 	return
 }
 
+// updateDeprecatedStatus adds deprecation status conditions to the subscription when present in the cache entry
+func (c *catalogHealthReconciler) updateDeprecatedStatus(ctx context.Context, sub *v1alpha1.Subscription) (bool, error) {
+	if c.sourceProvider == nil {
+		return false, nil
+	}
```

Comment:
Should we have some kind of logging or return a non-nil error here? I can see it being useful to look at the logs and get some insights for debugging any issues with deprecation status not being set.

My same question applies to all locations where we return a nil error.

Without more familiarity with the overall flow that OLM takes here I would lean more towards some kind of logging and keeping the nil error returns, but if anyone more familiar with this logic thinks it is better to do non-nil error returns I'm happy with that as well.

Reply:
Do we ever expect `c.sourceProvider` to be `nil`? If not, I'd suggest a `panic` here.

Reply:
I'm personally fine with the nil check, but I don't like the idea of this "failing" silently. I'd at the least have a log message.

Reply:
Since we don't at this moment know what kind of logging impact there would be from logging here, we will take this as a follow-up.

Reply:
Could we create an issue to investigate the logging and test changes discussed in this thread?